src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -42,9 +42,9 @@ class KalmanPosterior(TimeSeriesPosterior, abc.ABC):
 
     def __init__(
         self,
-        locations: np.ndarray,
-        states: _randomvariablelist._RandomVariableList,
         transition: GaussMarkovPriorTransitionArgType,
+        locations: Optional[Iterable[FloatArgType]] = None,
+        states: Optional[Iterable[randvars.RandomVariable]] = None,
     ) -> None:
```

Comment:
@pnkraemer Do you approve of this change of the ordering of the arguments?
This is necessary in order to also create empty `FilteringPosterior` objects. Since non-optional arguments may not follow optional ones. And the transition **always** has to be provided. Even for empty posteriors.

Reply:
Yes, I could live with that ;) However, do you think it would improve readability even more if at initialisation one would (need to) pass the empty lists? This might make it even more obvious to a code reader what happens, but that is just a guess. What do you think?

Reply:
I personally don't think that it would improve readability, or ease-of-use. But that's personal taste :)

Reply:
alright, then not :)